### PR TITLE
Changling power: regenerate; regrow limbs, organs

### PR DESCRIPTION
**Pull request by coiax**
:cl: coiax
del: Fleshmend no longer regrows lost limbs. 
add: Added free Regenerate power to changelings, which regrows all
external limbs and internal organs
(including the tongue if you lost your head). It costs 10 chemicals to
use.
add: Fleshmend now costs 20 chemicals, down from 25.
tweak: Changeling revive power name changed from Regenerate to Revive.
/:cl:
- Also makes fleshmend use a 0'd addtimer rather than a spawn.
- Thrown limbs now don't always land in the centre of the tile.

### CODE REVIEW

**Comment by Cheridan** [comment]
can i cut off a changeling's armblade yet and beat him with it?

**Comment by Armhulen** [comment]
if a changeling regenerates, all the lost limbs and organs should turn into hostile simplemobs that attack anyone nearby

**Comment by Core0verload** [comment]
Pointless nerf. :-1:

**Comment by coiax** [comment]
Fleshmend is already too potent, it heals about 70 of each damage, but that's not enough; we want it to regrow all organs and limbs, remove shrapnel and restore blood? I mean come on.

**Comment by kevinz000** [comment]
Terrible

**Comment by InsaneHyena** [comment]
Makes changelings literally unplayable.

**Comment by Bawhoppen** [comment]
Big balance change that would be overall good I'd say in the long run

**Comment by anconfuzedrock** [comment]
lesser form already does this though, including the organs.

**Comment by vulpineryth** [comment]
I would keep limb regen on Revive, otherwise regenerate will be a tax since revive would be pointless if you die again from not having a head or blood.

**Comment by coiax** [comment]
Lings don't die from not having a head, strangely enough. Apart from the lack of tongue, they're not inconvenienced.

**Comment by coiax** [comment]
@ChangelingRain I have rebalanced based on your feedback; is this now good?

**Comment by vulpineryth** [comment]
@coiax Yeah but nothing screams ling like a headless body walking around on one foot

File: code/game/gamemodes/changeling/powers/regenerate.dm [hunk 1]
```diff
@@ -0,0 +1,32 @@
+/obj/effect/proc_holder/changeling/regenerate
+	name = "Regenerate"
+	desc = "Allows us to regrow and restore missing external limbs, and \
+		vital internal organs, as well as removing shrapnel and restoring \
+		blood volume."
+	helptext = "Will alert nearby crew if any external limbs are \
+		regenerated. Can be used while unconscious."
+	chemical_cost = 10
+	dna_cost = 0
```
Comment (ChangelingRain) [inline code/game/gamemodes/changeling/powers/regenerate.dm]: should probably have `always_keep = TRUE`

File: code/game/gamemodes/changeling/powers/regenerate.dm [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+				reform, making a loud, grotesque sound!</span>",
+				"<span class='userdanger'>Your limbs regrow, making a \
+				loud, crunchy sound and giving you great pain!</span>",
+				"<span class='italics'>You hear organic matter ripping \
+				and tearing!</span>")
+			H.emote("scream")
+			H.regenerate_limbs(1)
+
+		CHECK_DNA_AND_SPECIES(H)
+		H.dna.species.on_species_gain(H, H.dna.species)
```
Comment (phil235) [inline code/game/gamemodes/changeling/powers/regenerate.dm]: @coiax  whyyyyy

**Pull request merged**